src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,75 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
+    {% spaceless %}
+        {% from _self import render_array %}
+
+        {% if not inline %}
+            <ul>
         {% else %}
-            [{{ key }} => {{ val }}]
+            [
         {% endif %}
 
-        {% if not loop.last and not inline %}
-            <br>
+        {% for key, val in value %}
+            {% if val is not iterable and not inline %}
+                <li>
+            {% endif %}
+
+            {% if options.display in ['both', 'keys'] or options.display is not defined %}
+                {% if options.key_translation_domain is same as(false) or options.key_translation_domain is not defined %}
+                    {{ key }}
+                {% else %}
+                    {% if options.key_translation_domain is same as(true) %}
+                        {% set key_translation_domain = options.default_translation_domain %}
+                    {% elseif options.key_translation_domain is same as(null) %}
+                        {% set key_translation_domain = null %}
+                    {% else %}
+                        {% set key_translation_domain = options.key_translation_domain %}
+                    {% endif %}
+
+                    {{ (options.translation_prefix ~ 'key.' ~ key)|trans({}, key_translation_domain) }}
+                {% endif %}
+            {% endif %}
+
+            {% if options.display == 'both' or options.display is not defined %}
+                &nbsp;=>&nbsp;
+            {% endif %}
+
+            {% if val is iterable %}
+                {{ render_array(val, inline, options) }}
+            {% else %}
+                {% if options.display in ['both', 'values'] or options.display is not defined %}
+                    {% if options.value_translation_domain is same as(false) or options.value_translation_domain is not defined %}
+                        {{ val }}
+                    {% else %}
+                        {% if options.value_translation_domain is same as(true) %}
```

Comment:
What happen if value_translation_domain is not defined ?

Reply:
Similar check on line 51 as for `key_translation_domain`